src/LaunchDarkly.ServerSdk/Internal/BigSegments/BigSegmentsInternalTypes.cs

```diff
@@ -23,10 +23,16 @@ internal static class BigSegmentsInternalTypes
             string.Format("{0}.g{1}", s.Key, s.Generation.Value);
 
         // This type is used when Evaluator is querying Big Segments
-        internal struct BigSegmentsQueryResult
+        internal readonly struct BigSegmentsQueryResult
         {
-            internal IMembership Membership { get; set; }
-            internal BigSegmentsStatus Status { get; set; }
+            internal IMembership Membership { get; }
+            internal BigSegmentsStatus Status { get; }
+
+            public BigSegmentsQueryResult(IMembership membership, BigSegmentsStatus status)
```

Comment:
I would strongly suggest bumping your C# language version to 10.0 and use `record struct` for these use cases:

```cs
internal readonly record struct BigSegmentsQueryResult(
    IMembership membership,
    BigSegmentsStatus status
);
```

You're going to run into the `IsExternalInit` issue because you're targeting older versions but this should be solvable by include this build time only dependency:

[IsExternalInit](https://www.nuget.org/packages/IsExternalInit)